### PR TITLE
Remove B-tree from index and constraint commands

### PR DESCRIPTION
**Pull request by Hunterness**
Updates docs to match the removal in neo4j: https://github.com/neo-technology/neo4j/pull/14780 (this should preferable be merged before that one as to not break docs)

### CODE REVIEW

**Review by OliviaYtterbrink — APPROVED** [review]

File: cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc [hunk 1]
```diff
@@ -29,16 +29,15 @@ There are multiple index types available:
 * Point index.
 * Range index.
 * Text index types.
-* B-tree. label:deprecated[]
 
 [NOTE]
 ====
-The deprecated B-tree index type has been replaced by more specific index types (Range, Point, and Text).
+The previous B-tree index type has been replaced by more specific index types (Range, Point, and Text).
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc]:
```suggestion
The B-tree index type has been replaced by more specific index types (Range, Point, and Text).
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 2]
```diff
@@ -162,6 +162,82 @@ Replaced by:
...
+----
+a|
+Node key and uniqueness constraints backed by B-tree indexes are removed.
+
+Replaced by:
+[source, cypher, role="noheader"]
+----
+CREATE CONSTRAINT ...
+----
+Constraints used for string properties will also require an additional text index to cover the string queries properly.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
Constraints used for string properties require an additional text index to cover the string queries properly.
```

File: cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc [hunk 3]
```diff
@@ -162,6 +162,82 @@ Replaced by:
...
+Constraints used for point properties will also require an additional point index to cover the spatial queries properly.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc]:
```suggestion
Constraints used for point properties require an additional point index to cover the spatial queries properly.
```

**Review by renetapopova — COMMENTED** [review]
Thanks, @Hunterness! My local build failed, so I couldn't verify the formatting. I guess that's because the corresponding PR in the `neo4j` repo must be merged first to be able to build that one. However, it looks good, I've just added some minor editorial comments.

**Comment by Hunterness** [comment]
> I guess that's because the corresponding PR in the neo4j repo must be merged first to be able to build that one.

No, I think this one should go in first as I should only be removing things (and merging neo4j first will break the docs)

**Comment by renetapopova** [comment]
Thanks, @Hunterness. I still can't verify it locally. My build has failed with the same errors as in TC.

**Comment by Hunterness** [comment]
> I still can't verify it locally. My build has failed with the same errors as in TC.

Looks like I need to rebase, will fix

**Comment by Hunterness** [comment]
@renetapopova The PR build is fixed, let's hope you can build it locally now as well :)

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc [hunk 4]
```diff
@@ -60,7 +56,7 @@ This section comprises a glossary of all the keywords -- grouped by category and
 |<<query-return, RETURN ... [AS]>>                    | Projecting   |  Defines what to include in the query result set.
 |<<query-set, SET>>                          | Writing     |  Update labels on nodes and properties on nodes and relationships.
 |<<administration-constraints-syntax-list, SHOW [ALL\|UNIQUE\|NODE [PROPERTY] EXIST[ENCE]\|REL[ATIONSHIP] [PROPERTY] EXIST[ENCE]\|[PROPERTY] EXIST[ENCE]\|NODE KEY] CONSTRAINT[S]>> | Schema | List constraints in the database, either all or filtered on type. Also allows `WHERE` and `YIELD` clauses.
-|<<administration-indexes-list-indexes, SHOW [ALL\|BTREE\|FULLTEXT\|LOOKUP] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on b-tree, fulltext or token lookup indexes. Also allows `WHERE` and `YIELD` clauses.
+|<<administration-indexes-list-indexes, SHOW [ALL\|FULLTEXT\|LOOKUP] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on b-tree, fulltext or token lookup indexes. Also allows `WHERE` and `YIELD` clauses.
```
Comment (renetapopova) [inline cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-indexes-list-indexes, SHOW [ALL\|FULLTEXT\|LOOKUP] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on fulltext, or token lookup indexes. Also allows `WHERE` and `YIELD` clauses.
```

**Review by renetapopova — COMMENTED** [review]
Builds fine now, thanks @Hunterness. I've found one more mention of `b-tree `that I think should be removed.

File: cypher/cypher-docs/src/docs/dev/glossary.asciidoc (hunk 4 above)
Comment (Hunterness) [reply cypher/cypher-docs/src/docs/dev/glossary.asciidoc]:
```suggestion
|<<administration-indexes-list-indexes, SHOW [ALL\|FULLTEXT\|LOOKUP\|POINT\|RANGE\|TEXT] INDEX[ES]>> | Schema | List indexes in the database, either all or filtered on fulltext, token lookup, point, range, or text indexes. Also allows `WHERE` and `YIELD` clauses.
```

It should probably have the new index types as well though

Comment (renetapopova) [reply cypher/cypher-docs/src/docs/dev/glossary.asciidoc]: Absolutely!

**Review by renetapopova — APPROVED** [review]
Thanks a lot, @Hunterness!

**Pull request merged**